service/history/replicationTaskExecutor.go

```diff
@@ -295,12 +295,19 @@ func (e *replicationTaskExecutorImpl) cleanupWorkflowExecution(ctx context.Conte
 	if err != nil {
 		return err
 	}
-	return e.deleteManager.DeleteWorkflowExecutionByReplication(
+
+	namespaceEntry := mutableState.GetNamespaceEntry()
+	if namespaceEntry.ActiveClusterName() == e.shard.GetClusterMetadata().GetCurrentClusterName() {
+		return serviceerror.NewInvalidArgument("Cannot cleanup workflows in active cluster by replication")
```

Comment:
This is not correct. There are cases where we do need to cleanup workflow in standby cluster. For example, standby side receive some replication task and was not able to apply, it would connect to active side directly, and when active side return NOT_FOUND, standby side would need to cleanup the workflow because otherwise those workflow will stay forever.

Reply:
This is a sanity check to make sure this cleanup only happen from standby clusters.

Reply:
You can remove this as we ensure this replication executor won't be executed in active cluster (at ns level).

Reply:
Removed.